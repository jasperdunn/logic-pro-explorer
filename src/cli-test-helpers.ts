import { execSync } from 'child_process'
import { resolve } from 'path'

export function runCommand(command: string): string {
  return execSync(`node ${resolve(__dirname, '../dist/src/index.js')} ${command}`).toString()
}
