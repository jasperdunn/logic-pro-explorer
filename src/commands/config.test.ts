import { runCommand } from '../cli-test-helpers.js'

describe('config command', () => {
  test('should display the help message', () => {
    expect(runCommand('config -h')).toMatchSnapshot()
  })
})
