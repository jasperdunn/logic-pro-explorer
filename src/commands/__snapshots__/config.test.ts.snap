// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config command > should display the help message 1`] = `
"Usage: logic-pro-explorer config [options]

Set configuration options.

Options:
  -p, --project-directory <path>    Set the directory to scan for Logic
                                    projects.
  -c, --component-directory <path>  Set the directory to scan for components.
  -r, --reset                       Reset the configuration to the default
                                    values.
  -h, --help                        display help for command
"
`;
